api/v1alpha1/search_types.go

```diff
@@ -0,0 +1,199 @@
+
+	// +optional
+	// Customization for search deployments
+	Deployments SearchDeployments `json:"deployments,omitempty"`
+
+	// +optional
+	// Specifies deployment replication for improved availability. Options are: Basic and High (default)
+	AvailabilityConfig AvailabilityType `json:"availabilityConfig,omitempty"`
+
+	// +optional
+	// Control list of Kubernetes resources indexed by search-collector
+	AllowDenyResources FilterSpec `json:"allowDenyResources,omitempty"`
+
+	// +optional
+	// Kubernetes secret name containing user provided db secret
+	// Secret contains connection url, certificates
+	CustomDbConfig string `json:"customDbConfig,omitempty"`
+}
+
+type SearchDeployments struct {
+	// +optional
+	// Configuration for DB Deployment
+	Database DeploymentConfig `json:"database,omitempty"`
+
+	// +optional
+	// Configuration for indexer Deployment
+	Indexer DeploymentConfig `json:"indexer,omitempty"`
+
+	// +optional
+	// Configuration for collector Deployment
+	Collector DeploymentConfig `json:"collector,omitempty"`
+
+	// +optional
+	// Configuration for api Deployment
+	API DeploymentConfig `json:"api,omitempty"`
+
+	// +optional
+	// Configuration for addon installed collector Deployment
+	RemoteCollector DeploymentConfig `json:"remoteCollector,omitempty"`
+}
+
+type DeploymentConfig struct {
+	// +optional
+	// Number of pod instances for deployment
+	ReplicaCount int `json:"replicaCount,omitempty"`
+	// +optional
+	// Compute Resources required by deployment
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+	//Image_override
+	ImageOverride string `json:"imageOverride,omitempty"`
+
+	// +optional
+	//ImagePullSecret
+	ImagePullSecret string `json:"imagePullSecret,omitempty"`
+
+	//ImagePullPolicy
+	// +optional
+	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
+
+	// NodeSelector to schedule on nodes with matching labels
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	//Proxy config , if remote collectors need to override
+	// +optional
+	ProxyConfig map[string]string `json:"proxyConfig,omitempty"`
+}
+
+type StorageSpec struct {
+	// +optional
+	// name of the storage class
+	StorageClassName string `json:"storageClassName,omitempty"`
+	// +optional
+	// storage capacity
+	Size *resource.Quantity `json:"size,omitempty"`
+}
+
+type FilterSpec struct {
+	// +optional
+	// Allowed resources from collector
+	AllowedResources ResourceListSpec `json:"allowedResources,omitempty"`
+
+	// +optional
+	// Denied resources from collector
+	DeniedResources ResourceListSpec `json:"deniedResources,omitempty"`
+}
+
+type ResourceListSpec struct {
+	//API Group names to be filtered
+	APIGroups []string `json:"apiGroups,omitempty"`
+	//Resource names to be filtered
+	Resources []string `json:"resources,omitempty"`
+	// +optional
+	//Cluster Labels this filter to be applied
+	ClusterLabels []string `json:"clusterLabels,omitempty"`
```

Comment:
1. Need to confirm that we want to use labels vs ClusterSets.
2. If using labels, we should implement as a [LabelSelector](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L1178-L1191) for consistency with K8s. It also adds more flexibility.